src/core_plugins/kibana/public/dashboard/panel/dashboard_panel.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { DashboardViewMode } from '../dashboard_view_mode';
+import { PanelHeader } from './panel_header';
+
+export class DashboardPanel extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.embeddable = null;
+    this.embeddableHandler = null;
+    this.parentNode = null;
+  }
+
+  async componentWillMount() {
```

Comment:
hmm, this actually causes a lint error:
`error  Do not use setState in componentDidMount  react/no-did-mount-set-state`

Something that could be fixed with redux, but not sure how to improve it with this setup, rather than just overriding the eslint rule.

Reply:
Ah, you hit that because you made the lifecycle hook `async` and eslint still believes the `setState` is happening synchronously, I expect. Not sure how to best solve that, I think I'd either `eslint-disable` on that line or disable that eslint setting, I think. Or used promises instead of `await`.

(I don't think you need the `isMounted` check, though)

Reply:
(Oh, just now saw your comment re `isMounted`. I might have read the code wrong and you actually need it. Sorry if causing confusion 🙈)